src/goto-symex/build_goto_trace.cpp

```diff
@@ -92,7 +92,9 @@ exprt build_full_lhs_rec(
           id==ID_byte_extract_big_endian)
   {
     exprt tmp=src_original;
-    assert(tmp.operands().size()==2);
+    DATA_INVARIANT(
+      tmp.operands().size() == 2,
+      "byte_extract_exprt should have two operands.");
```

Comment:
Let's please use `to_byte_extract_expr`, and use `byte_extract_exprt`'s API. Otherwise we duplicate consistency checks across the code base and risk getting out-of-sync.